web/src/stream_topic_history.ts

```diff
@@ -234,9 +276,36 @@ export function remove_messages(opts: {
         return;
     }
 
+    if (opts.topic_deleted) {
+        // topic_deleted is true only when propagate_mode is
+        // set to change_all and here we can be sure that the
+        // topic will no longer exist, so we will remove the
+        // topic completely.
+        history.topics.delete(topic_name);
+        return;
+    }
+
     // Update max_message_id in topic
     if (existing_topic.message_id <= max_removed_msg_id) {
         const msgs_in_topic = message_util.get_messages_in_topic(stream_id, topic_name);
+
+        if (!msgs_in_topic.length) {
+            get_latest_message_in_topic(stream_id, topic_name, (latest_message_id) => {
+                if (!latest_message_id) {
```

Comment:
While testing, I noticed sometimes that the `latest_message_id` returned here was the message which was moved to the new topic. I cannot reproduce this always, so not sure why is this happening, but will investigate again.

Reply:
I did some print debugging for this and found that the `Message` object was not updated as expected. I printed the `Message` object after the message is updated in database in `update_message` codepath and the `recipient_id` points to the new stream, but when I print the `Message` object in the message fetch code the `recipient_id` field is set to old stream.